tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/server/AbstractCodeActionTest.java

```diff
@@ -71,9 +75,48 @@ protected void performTest(Project project, String moduleName, TestCodeActionCon
 		if (tcac.getAssertCodeActions() != null) {
 			tcac.getAssertCodeActions().apply(result);
 		} else {
-			String resultStr = Strings.toString(getStringLSP4J()::toString3, result);
+			String resultStr = result.stream()
+					.map(cmdOrAction -> getStringLSP4J().toString3(cmdOrAction))
+					.collect(Collectors.joining("\n-----\n"));
 			assertEquals(tcac.getExpectedCodeActions().trim(), resultStr.trim());
 		}
 	}
 
+	/** Like {@link TestCodeActionConfiguration}, but non-empty ranges can be defined. */
+	public static class TestCodeActionConfiguration2 extends TestCodeActionConfiguration {
```

Comment:
Why `...2`? Maybe there would be a more telling name (sometimes we use `N4JS...`).